packages/x-charts/src/BarChart/BarElement.tsx

```diff
@@ -103,6 +110,7 @@ function BarElement(props: BarElementProps) {
       fill: color,
       skipAnimation,
       layout,
+      'data-focused': isFocused || undefined,
```

Comment:
[Base UI](https://base-ui.com/react/components/menu) uses `data-highlighted`, but that might conflict with our current highlighted logic. 

<img width="1719" height="1380" alt="Image" src="https://github.com/user-attachments/assets/944eca11-10a3-4801-ba82-4a6995dc2cf8" />